tests/dbt/test_graph.py

```diff
@@ -504,6 +504,24 @@ def test_update_node_dependency_test_not_exist():
         assert nodes.has_test is False
 
 
+def test_tag_selected_node_test_exist():
+    project_config = ProjectConfig(
+        dbt_project_path=DBT_PROJECTS_ROOT_DIR / DBT_PROJECT_NAME, manifest_path=SAMPLE_MANIFEST
+    )
+    profile_config = ProfileConfig(
+        profile_name="test",
+        target_name="test",
+        profiles_yml_filepath=DBT_PROJECTS_ROOT_DIR / DBT_PROJECT_NAME / "profiles.yml",
+    )
+    dbt_graph = DbtGraph(project=project_config, profile_config=profile_config, select=["tag:test_tag"])
+    dbt_graph.load_from_dbt_manifest()
+
+    for _, node in dbt_graph.filtered_nodes.items():
```

Comment:
@edgga could we also do an assert in the len of `dbt_graph.filtered_nodes`, just to be on the safe side?

I usually like to avoid logic  such as for loops in tests - since that increases the chances for bugs - but it's okay for this test case, as long as we confirm the for won't be iterating in a empty list :)